tests/syscall/mount/mount-immutables.bats

```diff
@@ -0,0 +1,596 @@
+}
+
+# Testcase #9.
+#
+# Ensure that a read-write immutable mount *can* be masked by
+# a new read-only mount on top of it.
+@test "ro mount on top of immutable rw mount" {
+  if [[ $skipTest -eq 1 ]]; then
+    skip
+  fi
+  local syscont=$(docker_run --rm ${CTR_IMG_REPO}/ubuntu:latest tail -f /dev/null)
+  local immutable_rw_mounts=$(list_container_rw_dir_mounts ${syscont} "0" "/")
+  run empty_list ${immutable_rw_mounts}
+  [ "$status" -ne 0 ]
+  echo ${immutable_rw_mounts} > /work_list
+
+  for m in $immutable_rw_mounts; do
+
+    # skip /proc and /sys since these are special mounts (we have dedicated
+    # tests for remounting them). We also
+    if [[ ${m} =~ "/proc" ]] || [[ ${m} =~ "/proc/*" ]] ||
+         [[ ${m} =~ "/sys" ]] || [[ ${m} =~ "/sys/*" ]] ||
+         [[ ${m} =~ "/dev" ]] || [[ ${m} =~ "/dev/*" ]]; then
+      continue
+    fi
+
+    # This should pass (mount is read-write)
+    docker exec ${syscont} sh -c "touch ${m}"
+    [ "$status" -eq 0 ]
+
+    printf "\nmounting tmpfs (ro) on top of immutable rw mount ${m}\n"
+
+    docker exec ${syscont} sh -c "mount -t tmpfs -o ro,size=100M tmpfs ${m}"
+    [ "$status" -eq 0 ]
+
+    # This should fail (tmpfs mount is read-only)
+    docker exec ${syscont} sh -c "touch ${m}"
+    [ "$status" -ne 0 ]
+
+    docker exec ${syscont} sh -c "umount ${m}"
+    [ "$status" -eq 0 ]
+
+    # This should pass (mount is read-write)
+    docker exec ${syscont} sh -c "touch ${m}"
+    [ "$status" -eq 0 ]
+  done
+
+  docker_stop ${syscont}
+}
+
+# Testcase #10.
+@test "don't confuse inner priv container mount with immutable mount" {
+  if [[ $skipTest -eq 1 ]]; then
+     skip
+  fi
+  local syscont=$(docker_run --rm ${CTR_IMG_REPO}/alpine-docker-dbg tail -f /dev/null)
+  local immutable_ro_mounts=$(list_container_ro_mounts ${syscont} "0" "/")
+  run empty_list ${immutable_ro_mounts}
+  [ "$status" -ne 0 ]  
+  
+  local linux_libmod_mount=$(echo $immutable_ro_mounts |  tr ' ' '\n' | grep "lib/modules")
+
+  [ -n "$linux_libmod_mount" ]
+
+  docker exec -d ${syscont} sh -c "dockerd > /var/log/dockerd.log 2>&1"
+  [ "$status" -eq 0 ]
+
+  wait_for_inner_dockerd ${syscont}
+
+  docker exec ${syscont} sh -c "docker run --privileged -d --name inner ${CTR_IMG_REPO}/ubuntu:latest tail -f /dev/null"
+  [ "$status" -eq 0 ]
+
+  # In the inner container, create a dir and mountpoint whose name matches that
+  # of an immutable sysbox mount
+  docker exec ${syscont} sh -c "docker exec inner sh -c \"mkdir -p $linux_libmod_mount\""
+  [ "$status" -eq 0 ]
+
+  docker exec ${syscont} sh -c "docker exec inner sh -c \"mount --bind $linux_libmod_mount $linux_libmod_mount\""
+  [ "$status" -eq 0 ]
+
+  # Verify we can write, remount, and umount the newly created mountpoint without problem
+  docker exec ${syscont} sh -c "docker exec inner sh -c \"touch $linux_libmod_mount\""
+  [ "$status" -eq 0 ]
+
+  docker exec ${syscont} sh -c "docker exec inner sh -c \"mount -o remount,bind,ro $linux_libmod_mount\""
```

Comment:
I would add the corresponding `touch ${mnt}` checks after each of the mount steps.